### ISSUE TITLE
Curl errors are not logged or returned to the caller

### ISSUE THREAD

**Issue by LoukKamp** [domoticz/domoticz#6147]
Today, I simulated a power fail of my Domoticz server to test if everything comes back when the power is restored. 

It all seemed to work fine, but after 3 hours, the Netatmo interface started to complain about being unable to get refresh the tokens. Strange, because I was able to log in to the system and show Domiticz pages through the web server interface. I was also able to get a reply from Netatmo when I tried to access the API portal from my laptop (on the same network). Then I tried to use curl from the Domoticz server, and it returned an error: "No path to host". I'm not sure what caused this, but I guess something with routing. I didn't have time to figure this out.

After some detective work, I found that the call to _HTTPClient::POST_ is defined in  _httpclient/HTTPClient.cpp_ where _HTTPClient::POSTBinary_ is called to do the call to curl.

When something went wrong, the only error that is logged is _CURLE_HTTP_RETURNED_ERROR_. All other errors are pushed to the (return) header stack.

So, I changed Netatmo.cpp to use the call to POST with 5 parameters (one extra for the return headers). This returned:

`HTTP/1.1 7 Couldn't connect to server
`
Although this is more than nothing, it doesn't point to the real cause (No Path To Host).

Would it be possible to improve this exception handling, maybe by upgrading the referenced libraries? Or maybe give the OS a bit more time at startup to initialize by delaying Domoticz a bit?

**Comment by gizmocuz** [comment]
By upgrading the referenced libraries? We are already making use of the latest libraries and headers.
Giving the OS more time, well Domoticz should only start when the network is active.

It could be various things, like not able to query the DNS server, your router may not be ready yet.

Surely after 5 minutes this should all be operational and with the next poll interval the Netamo server can be reached?
Not sure how frequently the Netamo hardware in Domoticz polls, but if there is an error getting results, you could let it poll after 5 minutes again?

**Comment by LoukKamp** [comment]
I already assumed, that Domoticz was using the latest version; just wanted to be sure. Maybe I was misled by the text in the error message: "HTTP/1.1", but I realized later that this version number is hard coded in _httpclient/HTTPClient.cpp_. I was just looking for a reason why the curl command showed a different reason than Domoticz (using the curl library).

This exception is not caused by Domoticz, but probably somewhere in the OS. I needed to reboot the system to get rid of it (the issue doesn't recover by itself after some time). Maybe there is a more sophisticated way, but I did have the time to investigate this, because there is no data coming in as long as there is no communication possible.

My best guess would be a lock issue during system startup. But I need more time to try some options. The problem is that I don't have a second system to experiment with this and I don't want big holes in my data.
